container/Containerfile.gitpod

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# SSH credentials:
+#   root@password
+
+FROM docker.io/gitpod/workspace-full:2023-10-06-16-22-14@sha256:76d3041cc7a2caa00d6f4610ace0e15009c361515f3d5d9ee6690e4019adcfd4
+
+# Switch to root user to install dependencies and configure sshd
+USER root
+
+# Install dependencies
+RUN DEBIAN_FRONTEND="noninteractive" apt-get update && \
+    DEBIAN_FRONTEND="noninteractive" apt-get install -y --no-install-recommends \
+        # Development dependencies \
+        dos2unix \
+        git \
+        locales-all \
+        rsync \
+        tar \
+        ssh \
+        tzdata \
+        # Build dependencies \
+        build-essential \
+        ccache \
+        cmake \
+        curl \
+        default-jdk \
+        gcc \
+        g++ \
+        gdb \
+        make \
+        python3 \
+        # Celix dependencies \
+        civetweb \
+        libavahi-compat-libdnssd-dev \
+        libcivetweb-dev \
+        libcpputest-dev \
+        libcurl4-openssl-dev \
+        libczmq-dev \
+        libffi-dev \
+        libjansson-dev \
+        libxml2-dev \
+        libzip-dev \
+        rapidjson-dev \
+        uuid-dev && \
+    apt-get clean
+
+RUN ( \
+    echo 'Port 2233'; \
+    echo 'LogLevel INFO'; \
+    echo 'PermitRootLogin yes'; \
+    echo 'PasswordAuthentication yes'; \
+    echo 'Subsystem sftp /usr/lib/openssh/sftp-server'; \
+  ) > /etc/ssh/sshd_config_celix \
+  && mkdir /run/sshd
+
+RUN yes password | passwd root
```

Comment:
I am not sure if setting the root password is wise. Especially when also running this at gitpod.io, to be honest I do not know the technical details I think this could be an issue.

I think in this case I rather see a gitpod and local ubuntu develop containerfile.

Reply:
Good point! Since we are indeed not sure how Gitpod is running their container (I surely hope not with `--net=host`) it is better to not set a password for the root user.

I updated the pull-request to re-introduce the Ubuntu containerfile.